spec/lib/hubspot/company_properties_spec.rb

```diff
@@ -14,158 +14,325 @@
+          expect(response).to be nil
         end
       end
     end
 
-    describe '.update!' do
-      context 'with no valid parameters' do
-
-        it 'should return nil ' do
-          expect(Hubspot::CompanyProperties.update!(params['name'], {})).to be(nil)
+    describe ".update!" do
+      it "updates the company property" do
+        VCR.use_cassette "company_properties/update_property" do
+          name = "fax_number"
+          label = "fax number label"
+          new_label = "new fax number label"
+
+          Hubspot::CompanyProperties.create!({
+            groupName: "companyinformation",
+            label: label,
+            name: name,
+            type: "string",
+          })
+
+          update_params = {
+            groupName: "companyinformation",
+            label: new_label,
+            type: "string",
+          }
+
+          response = Hubspot::CompanyProperties.update!(name, update_params)
+
+          assert_hubspot_api_request(
+            :put,
+            "/companies/v2/properties",
+            body: {
+              groupName: "companyinformation",
+              label: new_label,
+              options: [],
+              type: "string",
+            }
+          )
+          expect(response["label"]).to eq(new_label)
+
+          Hubspot::CompanyProperties.delete!(name)
         end
       end
 
-      context 'with mixed parameters' do
-        cassette 'company_properties/update_property'
+      context "with no valid parameters" do
+        it "returns nil" do
+          company_property_name = "fax_number"
+          params = {}
 
-        it 'should return the valid parameters' do
-          params['description']       = 'What is their favorite flavor?'
-          valid_params['description'] = params['description']
+          response = Hubspot::CompanyProperties.update!(
+            company_property_name,
+            params
+          )
 
-          response = Hubspot::CompanyProperties.update!(params['name'], params)
-          valid_params.each { |k, v| expect(response[k]).to eq(v) }
+          expect(response).to be nil
         end
       end
     end
 
-    describe '.delete!' do
-      let(:name) { params['name'] }
+    describe ".delete!" do
+      it "deletes the company property" do
+        VCR.use_cassette("company_properties/delete_property") do
+          name = "fax_number"
 
-      context 'with existing property' do
-        cassette 'company_properties/delete_property'
+          Hubspot::CompanyProperties.create!({
+            groupName: "companyinformation",
+            name: name,
+            type: "string",
+          })
 
-        it 'should return nil' do
-          expect(Hubspot::CompanyProperties.delete!(name)).to eq(nil)
+          response = Hubspot::CompanyProperties.delete!(name)
+
+          assert_hubspot_api_request(
+            :delete,
+            "/companies/v2/properties/named/#{name}"
+          )
+
+          expect(response).to be nil
```

Comment:
Is it worth having an expectation on the status code?

Reply:
This method is returning just `nil`, so I would need to alter the return value of this function to check the status code. I think that's worth doing but I'd prefer to make that change in a separate PR.